src/datapoints-graph/charts/chart-realtime.service.ts

```diff
@@ -149,6 +222,58 @@ export class ChartRealtimeService {
       seriesMatchingDatapoint.data = this.removeValuesBeforeTimeRange(
         seriesMatchingDatapoint
       );
+
+      alarmsAndEvents.forEach((item) => {
+        if (
+          typeof seriesMatchingDatapoint.data !== 'object' ||
+          seriesMatchingDatapoint.data === null
+        ) {
+          return;
+        }
+        const renderType: DatapointChartRenderType =
+          datapoint.renderType || 'min';
+        const dp: DatapointWithValues = {
+          ...datapoint,
+          values: seriesMatchingDatapoint.data as {
+            [date: string]: { min: number; max: number }[];
+          },
+        };
+
+        if (isEvent(item)) {
+          const newEventSeries =
+            this.echartsOptionsService.getAlarmOrEventSeries(
+              dp,
+              renderType,
+              false,
+              [item],
+              'event',
+              item.id
+            );
+          allDataSeries.push(...newEventSeries);
+        } else if (isAlarm(item)) {
+          const alarmExists = allDataSeries.some((series: { data: any[] }) =>
+            series.data.some((data) => data[0] === item.creationTime)
+          );
+          if (alarmExists) {
+            const alarmSeries = allDataSeries.find((series: { data: any[] }) =>
+              series.data.some((data) => data[0] === item.creationTime)
+            );
+            // Instead of updating the existing alarm series, it would be easier if we remove all existing series for the current time range and add them again.
+            allDataSeries.splice(allDataSeries.indexOf(alarmSeries), 1);
```

Comment:
"Instead of updating the existing alarm series, it would be easier if we remove all existing series for the current time range and add them again." why don't you do it? :D Array manipulation is always risky.